fleetspeak/src/server/mysql/mysql.go

```diff
@@ -60,6 +62,29 @@ func (d *Datastore) Close() error {
 	return d.db.Close()
 }
 
+// toClientID converts a uint64 into a ClientID.
+func toClientID(id uint64) common.ClientID {
+	if id == 0 {
+		return common.ClientID{}
+	}
+	d := [8]byte{}
+	binary.LittleEndian.PutUint64(d[:], id)
+	r, err := common.BytesToClientID(d[:])
+	if err != nil {
+		// Should never happen, since d is hardcoded to the right length.
+		log.Fatalf("Unable to parse client id [%x]: %v", err)
+	}
+	return r
+}
+
+// fromClientID converts a ClientID into a uint64.
+func fromClientID(id common.ClientID) uint64 {
```

Comment:
I am not sure abut the name of this function. If you read its usage (`fromClientId(client_id)`) it is no immediately known what is going: from client id into what? You have to lookup the definition to figure this out. And you even need an extra comment to fully describe what this function does. I think that something more descriptive like `FromClientIDToUint64` or `Uint64FromClientID`, while being lengthier, is much better to understand and read.

I am not sure how it goes with "Go spirit" though. I know that Go tends to be rather concise and this feels more javastic (or even objective-cish where you cram entire sentences into identifiers). So treat it just as a suggestion.

`toClientID` is fine and descriptive enough on its own. However, if you decide to rename `fromClientID` then for consistency reasons you should adjust `toClientID` accordingly.

Reply:
It is unexported, so I don't think we should work too hard to make it unambiguous, but it is used across a few files, so I wanted to give a bit of hint. Also, since they are conceptually simple and used in expressions, I'd really hate to make the names much longer, so that is what I ended up with, but I'm open to other suggestions.

I actually don't see toClientID and fromClientID as being different in terms of ambiguity. In some sense, in both cases you may need to check the signature to know what is going on (go doesn't do overloaded methods, and inferring the input type isn't necessary easier that inferring the output type).

How about toClientID -> fromDBClientID and fromClientID->toDBClientID?

Reply:
> It is unexported

Okay, I was not aware that in a package in Go is not roughly equivalent to file as usually happens in other languages. In that case I am inclined to accept current naming conventions.

> I actually don't see toClientID and fromClientID as being different in terms of ambiguity. In some sense, in both cases you may need to check the signature to know what is going on (go doesn't do overloaded methods, and inferring the input type isn't necessary easier that inferring the output type).

Maybe. I am probably just biased here since most programming languages allow overloading based on function parameters but not on the return type so `toClientID` feels natural whereas `fromClientID` does not. But the longer I think about that, the more convinced I am that you are right.

> The important bit isn't that we get/have a uint64 - we aren't about to do math on them, and I actually considered using a binary(8) instead.

In Rust or Haskell I would just [newtype](https://github.com/rust-unofficial/patterns/blob/master/patterns/newtype.md) them. But I don't think this is necessary as long as `ClientID` is used everywhere except functions that directly interface with the database.